### DISCUSSION TITLE
How to save mc chat color codes into logfile

### DISCUSSION THREAD

**Discussion by udu3324** [ORelio/Minecraft-Console-Client#1807] [Q&A] [answered]
How can I config MCC to save color codes (example: §b) into its logfile? I made logtofile true and the logfile has no chat color codes.

MinecraftClient.ini
```
[Logging]
# Only affect the messages on console.
debugmessages=false                # Please enable this before submitting bug reports. Thanks!
chatmessages=true                  # Show server chat messages
warningmessages=true               # Show warning messages
errormessages=true                 # Show error messages
infomessages=true                  # Informative messages (i.e Most of the message from MCC)
filtermode=blacklist               # blacklist OR whitelist. Blacklist hide message match regex. Whitelist show message match regex
logtofile=true                     # Write log messages to file
logfile=latest.log                 # Log file name
prependtimestamp=false              # Prepend timestamp to messages in log file
```

The text it saved:
```
                             Links
                 Website, Store, Twitter, Discord
                           Commands
                   Friends, Parties, Extra help
```

What it looks like in console:
![image](https://user-images.githubusercontent.com/47045986/139183594-439ace73-49a4-4353-baa1-fc2d23e3ce2b.png)

**Comment by ORelio** [comment, marked as answer]
Currently not possible. Created issue #1810 (enhancement request) for you.

**Reply by ORelio** [reply]
The requested feature is now implemented thanks to @Daenges's contribution in #1811.
On the latest [development build](https://ci.appveyor.com/project/ORelio/minecraft-console-client/build/artifacts) you can now set `savecolorcodes=true` under `[Logging]` to keep formatting codes.
